src/main/scala/zio/s3/Live.scala

```diff
@@ -130,9 +130,17 @@ final class Live(unsafeClient: S3AsyncClient) extends S3.Service {
     bucketName: String,
     key: String,
     content: ZStream[R, Throwable, Byte],
-    options: MultipartUploadOptions = MultipartUploadOptions()
-  ): ZIO[R, S3Exception, Unit] =
+    options: MultipartUploadOptions
+  )(parallelism: Int): ZIO[R, S3Exception, Unit] =
     for {
+      _        <- ZIO.cond(
```

Comment:
I would have the same validation for the part size without any additional types.

Reply:
i follow your suggestion